### ISSUE TITLE
Question about the pre-trained checkpoints results

### ISSUE THREAD

**Issue by ElonJustin7** [deeplearning-wisc/cider#5]
Hi author,

Thanks for the great work! These days I'm following this work and using this repo.
But I have a question about the pre-trained checkpoints results:

Using the provided pre-trained checkpoints, I can get nearly the same result as the paper presents; but when I use the provided script to trained CIFAR100 from scratch, I got a worse result that reported(Average 10% worse). Could this be due to the different environment or hyperparameters? I'm confused. Here is my hyperparameters and environment settings.

    --in-dataset CIFAR-100\
    --id_loc datasets/CIFAR100 \
    --gpu 2 \
    --model resnet34 \
    --loss cider \
    --epochs 500 \
    --proto_m 0.5 \
    --feat_dim 128 \
    --w 2 \
    --batch-size 512 \
    --cosine    \
Python==3.7.13
numpy==1.21.5
torch==1.12.1
torchvision==0.13.1

Thanks!

**Comment by alvinmingwisc** [comment]
Thanks for your comments! The provided checkpoint is intended to replicate the CIDER results shown in Table 1. We also present the average performance in Table 6 (Appendix C). Indeed, as observed in your experiments, the average FPR is around 10% worse when using default hyperparameters. We note that most hyperparameters (e.g., initial learning rate, learning rate schedule, number of epochs) are set to match those of SupCon (https://github.com/HobbitLong/SupContrast) for simplicity and to illustrate CIDER's effectiveness as a plug-in replacement. We have not conducted extensive hyperparameter tuning for CIFAR-100.

**Comment by ElonJustin7** [comment]
Thanks for your answer! And could you please provide the hyperparameter of the checkpoint that achieves the performance of Table 1 FPR40.85 in the CIFAR-100 experiment?

**Comment by alvinmingwisc** [comment]
Hi! The hyperparameter is the default one (avg FPR with α = 0.5 is around 44, as shown in the paper).  Btw, besides hyperparam tuning, there are several techniques to boost the performance, such as use KNN in conjunction with ReAct (https://arxiv.org/abs/2111.12797).
